clients/src/main/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumer.java

```diff
@@ -494,6 +495,16 @@ public void subscribe(Pattern pattern) {
         subscribeInternal(pattern, Optional.empty());
     }
 
+    @Override
+    public void subscribe(SubscriptionPattern pattern, ConsumerRebalanceListener callback) {
+
```

Comment:
Not stated in the KIP I believe, but I would say that we should log at least a warn indicating that this is not supported with the legacy protocol (similar to what the KIP states for logging a warn for the `enforceRebalance` that is not supported with the new protocol, and that we already do [here](https://github.com/apache/kafka/blob/055ff2b831193f5935f9efc2f7809f853f63de5f/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java#L1187))